CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.10)
+
+if(WIN32)
+	# Copy share library into build directory
+	set(WASM_TOOLS_INSTALL_COMMAND ${CMAKE_COMMAND} -E copy_if_different
+		${CMAKE_CURRENT_SOURCE_DIR}/target/release/libwasmtools.dll
+		${CMAKE_BINARY_DIR})
+else()
+	# TODO linux
+	set(WASM_TOOLS_INSTALL_COMMAND "")
+endif()
+
+include(ExternalProject)
+ExternalProject_Add(
+	wasm-tools-crate
+	DOWNLOAD_COMMAND ""
+	CONFIGURE_COMMAND ""
+	INSTALL_COMMAND ${WASM_TOOLS_INSTALL_COMMAND}
+	BUILD_COMMAND cargo build --release -p wasm-tools-c-api
+	BINARY_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+	LOG_BUILD ON
+	BUILD_ALWAYS ON)
+add_library(wasm-tools INTERFACE)
+add_dependencies(wasm-tools wasm-tools-crate)
+
+if(WIN32)
+	target_link_libraries(wasm-tools INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/target/release/libwasmtools.dll.lib)
+else()
+	# TODO linux
+endif()
+
+target_include_directories(wasm-tools INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/crates/c-api/include)
```

Comment:
I thought exposing the bindings to other code through the Cmake ecosystem would be more convenient, I personally used ExternalProject_Add when building before so anyone using the library would probably use the same solution. Since there's also a Cargo.toml I thought developers would recognize one is used for Rust and the other for C bindings

Reply:
Though I can see the value of moving the CMakeLists.txt into `crates/c-api`, it would be the same other than different relative paths